### PR TITLE
build: Move to es2021, fix some typing problems

### PR DESCRIPTION
**Pull request by martinpitt**
- [x] https://github.com/cockpit-project/bots/pull/7185

### CODE REVIEW

File: pkg/shell/router.jsx [hunk 1]
```diff
@@ -144,7 +144,7 @@ export function Router(callbacks) {
         }
 
         unique_id += 1;
-        const seed = (cockpit.transport.options["channel-seed"] || "undefined:") + unique_id + "!";
+        const seed = (cockpit.transport.options["channel-seed"] || "undefined:") + unique_id.toString() + "!";
```
Comment (cockpituous) [inline pkg/shell/router.jsx:147]: This added line is not executed by any test.

**Comment by martinpitt** [comment]
[This failure repeats](https://cockpit-logs.us-east-1.linodeobjects.com/pull-21391-fe0cac92-20241206-075953-fedora-coreos-networking/log.html), needs investigation.

File: build.js [hunk 2]
```diff
@@ -32,7 +32,7 @@ const pkgOptions = {
     nodePaths,
     outbase: './pkg',
     outdir: "./dist",
-    target: ['es2020'],
+    target: ['es2021'],
```
Comment (jelly) [inline build.js:35]: This should be lib, we don't want to target es2021 in the end result of the transpilation https://github.com/cockpit-project/cockpit/pull/21276/commits/5f68dfee226e155feb241ebe2cf1757e55cac996

**Comment by martinpitt** [comment]
I debugged the failure. It's completely unrelated to the changes here, they just trigger 3x affected retries. Fixed in https://github.com/cockpit-project/bots/pull/7185

File: build.js (hunk 2 above)
Comment (martinpitt) [reply build.js:35]: Yes, but I don't want half of our code 2021 and half 2020. Let's stick to *one* standard.

Comment (jelly) [reply build.js:35]: Alright! Would have also liked a .eslintrc.json downgrade but we can do that later.

**Review by jelly — APPROVED** [review]
Cool, I guess this was all the trivial stuff? :)

**Comment by martinpitt** [comment]
@jelly not all of it, I spent an hour on this and thought "this is big enough for a CI run". I'm sure there's much more low-hanging fruit.

**Pull request merged**